providers/vsphere/provider/provider.go

```diff
@@ -0,0 +1,218 @@
+// Copyright (c) Mondoo, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package provider
+
+import (
+	"errors"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"go.mondoo.com/cnquery/providers-sdk/v1/vault"
+
+	"go.mondoo.com/cnquery/llx"
+	"go.mondoo.com/cnquery/providers-sdk/v1/inventory"
+	"go.mondoo.com/cnquery/providers-sdk/v1/plugin"
+	"go.mondoo.com/cnquery/providers-sdk/v1/upstream"
+	"go.mondoo.com/cnquery/providers/vsphere/connection"
+	"go.mondoo.com/cnquery/providers/vsphere/resources"
+)
+
+type Service struct {
+	runtimes         map[uint32]*plugin.Runtime
+	lastConnectionID uint32
+}
+
+func Init() *Service {
+	return &Service{
+		runtimes:         map[uint32]*plugin.Runtime{},
+		lastConnectionID: 0,
+	}
+}
+
+func (s *Service) ParseCLI(req *plugin.ParseCLIReq) (*plugin.ParseCLIRes, error) {
+	flags := req.Flags
+	if flags == nil {
+		flags = map[string]*llx.Primitive{}
+	}
+
+	conn := &inventory.Config{
+		Type: req.Connector,
+	}
+
+	// parse args
+	user := ""
+	port := 0
+	if len(req.Args) != 0 {
+		target := req.Args[0]
+		if !strings.Contains(target, "://") {
+			target = "ssh://" + target
```

Comment:
Something doesn't match for me. Here, we expect an ssh connection string. At other places in the code, we use an API. I'm not sure how this should look like here.

Reply:
This does not use a ssh or the scheme at all. It is just a helper to use Go''s built-in URL-parse for the following:

```
cnquery shell vsphere chris@vsphere.local@192.168.5.24
```

In vSphere land we specifically need to tackle cases with double-@ which works with the built-in one super neat. I add a comment to make this more clear and rename it to something more neutral